modules/apps/commerce/commerce-product-content-web/src/main/resources/META-INF/resources/product_publisher/render/list/entry/view.jsp

```diff
@@ -101,7 +103,7 @@ boolean hasMultipleCPSkus = cpContentHelper.hasMultipleCPSkus(cpCatalogEntry);
 
 			<div>
 				<c:choose>
-					<c:when test="<%= !hasMultipleCPSkus && (cpSku != null) %>">
+					<c:when test="<%= (!hasMultipleCPSkus && (cpSku != null)) && !hasOptions %>">
```

Comment:
As mentioned previously, this is mimicking a similar check elsewhere which checks for the existence of any option and not just "required" options. I'm not sure if this is preferable.

Reply:
Is the quick-add-to-cart behavior not going to function properly for non-required options? If answer is yes and that it will properly function, we only need to check required options.

Please check whether following method covers your use-case: `hasCPDefinitionRequiredCPDefinitionOptionRels`

Reply:
It does not, since we do similar behavior elsewhere. Would you like me to change both of those to required options only then or are we okay with the lack of consistency?